crates/weaver_forge/codegen_examples/expected_codegen/lib.rs

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! OpenTelemetry Semantic Convention Attributes
+//! DO NOT EDIT, this is an Auto-generated file from templates/registry/rust/lib.rs.j2
+
+use opentelemetry::{Key, KeyValue, StringValue};
+
+
+/// These attributes may be used to describe the client in a connection-based network interaction where there is one side that initiates the connection (the client is the side that initiates the connection). This covers all TCP network interactions since TCP is connection-based and one side initiates the connection (an exception is made for peer-to-peer communication over TCP where the "user-facing" surface of the protocol / API doesn't expose a clear notion of client and server). This also covers UDP network interactions where one side initiates the interaction, e.g. QUIC (HTTP/3) and DNS.
+pub mod client;
+/// This document defines the shared attributes used to report a single exception associated with a span or log.
+pub mod exception;
+/// This document defines semantic convention attributes in the HTTP namespace.
+pub mod http;
+/// These attributes may be used for any network related operation.
+pub mod net;
+
+/// A typed attribute key.
```

Comment:
Not a problem introduced in this PR, but just wanted to share that OTel Rust already is suffering from performance issues in its abstractions (https://github.com/open-telemetry/opentelemetry-rust/issues/1642), so another abstraction here is going to make it worse :(

We are still working through the issue in OTel Rust group, not a blocker for this PR !

Reply:
In theory, this generated code, particularly for the type-safe API, could lead to more optimized code than what might be achievable with a generic client SDK.

Reply:
@cijothomas I've added a discussion on performance and on how, in the long run, a code generation approach can lead to optimized versions of an SDK client for a given application.